src/components/ArticlePage/index.jsx

```diff
@@ -9,32 +9,50 @@ import RatingStarsBox from '../../components/RatingStarsBox';
 import HorizontalLine from '../../components/HorizontalLine';
 import TextAreaInput from '../../components/TextAreaInput';
 import CommentEntries from '../../components/CommentEntries';
-import { getOneArticle } from '../../redux/actions/articles';
+import { getOneArticle, rateArticle, getArticleAvgRating } from '../../redux/actions/articles';
+import { toast } from 'react-toastify';
 import './style.scss';
 
 
 export class SingleArticleView extends Component {
 
-  componentDidMount() {
-    const { fetchArticle, match } = this.props;
+  async componentDidMount() {
+    const { fetchArticle, getArticleAvgRating, match } = this.props;
     let articleId = match.params.articleId;
-    fetchArticle(articleId);
+    await fetchArticle(articleId);
+    getArticleAvgRating(articleId)
+  }
+
+  starClickHandle(item) {
+    let articleId = this.props.match.params.articleId;
+    const _this = this;
```

Comment:
@Princesso Is there a need to use `line 28` ? Provide context please. No consistency in the use of `this` keyword in this function.

Reply:
@ifiokeyo. Usually, functions that are declared with the function keyword have a different scope. The convention is to use named functions everywhere so in the error handler, the function has a different scope from its parent scope.

Using unnamed functions however does not change the scope so the this keyword can work with unnamed functions unlike named functions.